enhancements/ingress/host-network-bind-options.md

```diff
@@ -0,0 +1,266 @@
+        // stats is the port number which HAProxy process binds
+        // to expose statistics on it. Setting this field is generally not recommended.
+        // However in HostNetwork strategy, default stats port 1936 might
+        // be occupied by other processess
+        //
+        // +kubebuilder:validation:Optional
+        // +kubebuilder:validation:Minimum=1
+        // +kubebuilder:validation:Maximum=30000
+        // +kubebuilder:default:=1936
+        // +optional
+        Stats int32 `json:"stats,omitempty"`
+    }
+    ```
+
+The following example configures two ingress controllers with the HostNetwork strategy that can run on the same nodes without port conflicts:
+
+    ```yaml
+    ---
+    apiVersion: operator.openshift.io/v1
+    kind: IngressController
+    metadata:
+    name: example-1
+    namespace: openshift-ingress-operator
+    spec:
+    endpointPublishingStrategy:
+        type: HostNetwork
+    ---
+    apiVersion: operator.openshift.io/v1
+    kind: IngressController
+    metadata:
+    name: example-2
+    namespace: openshift-ingress-operator
+    spec:
+    endpointPublishingStrategy:
+        type: HostNetwork
+        hostNetwork:
+            bindingOptions:
+                ports:
+                    http: 11080
+                    https: 11443
+                    stats: 1937
+    ```
+### Validation
+
+Setting `spec.endpointPublishingStrategy.type: HostNetwork` and ignoring
+`spec.endpointPublishingStrategy.hostNetwork` or
+`spec.endpointPublishingStrategy.hostNetwork.bindingOptions` is valid and implies
+the default behavior, which is to use the default ports to bind to. 
+
+The API validates
+that any value provided for the ports defined in
+`spec.endpointPublishingStrategy.hostNetwork.bindingOptions` is greater than `1`, and lower than `30000` (to avoid conflicts with 30000-32767 nodePort services range).
+
+### User Stories
+
+#### As a cluster administrator, I need to set binding ports for the IngressController
+
+The cluster administrator can set the ingress controller's `spec.endpointPublishingStrategy.hostNetwork.bindingOptions`.
+
+### API Extensions
+
+This  proposal edits `IngressController.operator.openshift.io/v1`.
+
+### Implementation Details
+
+Implementing this needs changes in the following repositories:
+
+* openshift/api
+* openshift/cluster-ingress-operator
+
+OpenShift Cluster Ingress Operator, creates a deployment for Router with environment variables for port bindings which OpenShift Router already respects:
+`ROUTER_SERVICE_HTTP_PORT`, `ROUTER_SERVICE_HTTPS_PORT`, `STATS_PORT`.
+
+Also the deployments should not have ip:port conflicts for `SNI_PORT` and `NO_SNI_PORT`. There are multiple ways to achieve this:
+
+- The cluster ingress operator configures the router to use a unique loopback address for the sni/nosni frontends. This loopback address would be determined deterministically, for example by hashing the ingresscontroller’s name.
```

Comment:
nit: not sure about "determined deterministically",  I'm not an English grammar expert but maybe something like "the loopback address is configured automatically , for example ...